src/main.rs

```diff
@@ -402,6 +402,29 @@ fn start_vmm(cmd_arguments: ArgMatches, api_socket_path: &Option<String>) -> Res
     .map(|()| log::set_max_level(log_level))
     .map_err(Error::LoggerSetup)?;
 
+    let (api_socket_path, api_socket_fd) =
+        if let Some(socket_config) = cmd_arguments.value_of("api-socket") {
+            let mut parser = OptionParser::new();
+            parser.add("path").add("fd");
+            parser.parse(socket_config).unwrap_or_default();
+
+            if parser.is_set("fd") {
+                (
+                    None,
+                    Some(parser.get("fd").unwrap().parse::<RawFd>().unwrap()),
```

Comment:
Could you try if `fd` is set by actually doing `if let Some(fd) = parser.get("fd") {`. This will remove the need for calling `is_set()` and then calling `unwrap()`.
Additionally, please replace `parse().unwrap()` with `parse.map_err()`.

Reply:
Ah right, of course. Thanks!